### PR TITLE
Execute tests for net451 and/or dnx451

### PR DESCRIPTION
**Pull request by natemcmaster**
Also publishes to obj/ instead of .testpublish/

cc @JunTaoLuo @pranavkm

### CODE REVIEW

File: KoreBuild-dotnet/build/_xunit-test.shade [hunk 1]
```diff
@@ -34,37 +34,47 @@ projectFile=''
...
             var runnerFolder = Path.GetFullPath(Path.Combine(KoreBuildFolderPath, "build", "xunit.runner.console", "tools"));
             var xunitCoreFolder = Path.GetFullPath(Path.Combine(KoreBuildFolderPath, "build", "xunit.core", "build", "_desktop"));
             
             foreach (var framework in targetFrameworks)
             {
-                if (!framework.StartsWith("dnxcore"))
+                if (IsLinux)
+                {
+                    // Work around issue with testing in parallel on Mono.
+                    testArgs = " -parallel none";
```
Comment (BrennanConroy) [inline KoreBuild-dotnet/build/_xunit-test.shade]: Since you're changing this code, I don't think -parallel is a switch anymore

Comment (natemcmaster) [inline KoreBuild-dotnet/build/_xunit-test.shade]: `-parallel` is still an option on xunit.console.exe. Did you mean we don't need to always specify this on Linux?

Comment (BrennanConroy) [inline KoreBuild-dotnet/build/_xunit-test.shade]: Hmm, maybe I'm thinking running `dotnet test -parallel none` doesn't work

Comment (JunTaoLuo) [inline KoreBuild-dotnet/build/_xunit-test.shade]: `dotnet test` is used for tests targeting `dnxcore50` and is used in `_dotnet-test.shade`. Tests targeting dnx451 or net451 will be run using the xunit runner which requires the switch.

**Comment by JunTaoLuo** [comment]
:shipit: as we will need it before updating test TFMs. Will need to update some repos that depend on the relative paths of the tests such as https://github.com/aspnet/ServerTests/blob/dev/test/ServerComparison.FunctionalTests/Helpers.cs#L13

**Comment by natemcmaster** [comment]
The relative paths shouldn’t need updating. I’m just moving the test publish folder sideways, not up or down.

**Comment by JunTaoLuo** [comment]
You're right, they don't have to be updated due to *this* change. However they still need to be updated since they have not been correct since we moved away from dnx test which launched the tests in the project directory.

**Pull request merged**